ptvsd/_remote.py

```diff
@@ -27,64 +31,150 @@ def _pydevd_settrace(redirect_output=None, _pydevd=pydevd, **kwargs):
 # start_server() method).
 # Then move at least some parts to the appropriate modules.  This module
 # is focused on running the debugger.
+def enable_attach(address, redirect_output=True,
+                  _pydevd=pydevd, _install=install,
+                  on_attach=lambda: None, **kwargs):
+    host, port = address
+
+    def wait_for_connection(daemon, host, port):
+        debugger = get_global_debugger()
+        while debugger is None:
+            time.sleep(0.1)
+            debugger = get_global_debugger()
+
+        debugger.ready_to_run = True
+        server = create_server(host, port)
+        client, _ = server.accept()
+        daemon.start_session(client, 'ptvsd.Server')
+        print('before re_build_breakpoints')
+        daemon.re_build_breakpoints()
+        print('after re_build_breakpoints')
+        on_attach()
+        print('after on_attach')
+
+    daemon = _install(_pydevd,
+                      address,
+                      start_server=None,
+                      start_client=(lambda daemon, h, port: daemon.start()),
+                      singlesession=False,
+                      **kwargs)
+
+    connection_thread = threading.Thread(target=wait_for_connection,
+                                         args=(daemon, host, port),
+                                         name='ptvsd.listen_for_connection')
+    connection_thread.pydev_do_not_trace = True
+    connection_thread.is_pydev_daemon_thread = True
+    connection_thread.daemon = True
+    connection_thread.start()
+
+    _pydevd.settrace(host=host,
+                     stdoutToServer=redirect_output,
+                     stderrToServer=redirect_output,
+                     port=port,
+                     suspend=False)
+<<<<<<< HEAD
```

Comment:
@DonJayamanne I don't think this is merged correctly.

Reply:
Lol, made some last minute changes.